Nix improvements

dtomvan opened the pull request:
- **refactor: unfuck flake**
- **chore: nix flake lock**
- **refactor: unfuck install.sh**
- **refactor: make install script portable**
- **refactor: use install.sh in nix package**
- **chore: gitignore nix builds**
- **feat: add nixos module**

This should improve the Nix user's experience, by:
- Using modern practices
- Not using overlays
- Providing a NixOS module for easy installation

Also `install.sh` has some general improvements as collateral "damage" :)

Also it isn't required to duplicate code anymore!

Tested the nix build, tested the nixos module.

dtomvan committed:
refactor: unfuck flake

dtomvan committed:
chore: nix flake lock

dtomvan committed:
refactor: unfuck install.sh

dtomvan committed:
refactor: make install script portable

dtomvan committed:
refactor: use install.sh in nix package

dtomvan committed:
chore: gitignore nix builds

dtomvan committed:
feat: add nixos module

nikp123 commented on install.sh:
```diff
@@ -1,59 +1,75 @@
-    exit 1
+if [ ! -d "${PLYMOUTHLIBS_PATH}" ]; then
+    echo "Please install Plymouth (On Fedora, plymouth-plugin-script)" &&
+        exit 1
```
This seems like weird indentation, was this done on purpose?

nikp123 commented:
Nicely done.

~~One slight detail I'd like to add, that isn't in this repo but is a fix for NixOS. During shutdown, plymouth doesn't see the themes' font since it only exists in initram.~~

~~Adding that package to fonts.packages does fix that, so I think it's a safe thing to add to the NixOS module, if we're already going that route.~~

Also unrelated, but: What do flake-parts and systems do here exactly? I see they're quite popular flakes among in other projects but I've never used them personally.

nikp123 committed:
Add plymouth font to system fonts in NixOS modules

This is to fix an issue with the plymouth themes' font not being present during shutdown, causing it to not appear correctly.

nikp123 commented:
Nice, I can just add that fix myself.

nikp123 commented:
Also it seems that overlays get a bad rep for some reason when used in packages like this, never could figure out why

dtomvan commented:
> Also unrelated, but: What do flake-parts and systems do here exactly? I see they're quite popular flakes among in other projects but I've never used them personally.

- `flake-parts` makes the process of writing flakes more unified, modular and simpler
- `systems` is an input that can be overridden by consumer flakes to influence which systems are available under `packages` etc.
